unikraft/core/transform.go

```diff
@@ -33,6 +35,14 @@ func TransformFromSchema(ctx context.Context, props interface{}) (interface{}, e
 
 	if source, ok := c["source"]; ok {
 		core.source = source.(string)
+
+		if f, err := os.Stat(core.source); err == nil && f.IsDir() {
+			if uk != nil && uk.UK_BASE != "" {
+				core.path = utils.RelativePath(uk.UK_BASE, core.source)
+			} else {
+				core.path = core.source
+			}
+		}
```

Comment:
From looking at this code, I'm not sure whether not setting `core.path` due to a non-nil `err` would cause any issue.
IMO that intention could be better conveyed by handling the error, with the cases that make we deem acceptable explicitly defined:

```golang
	f, err := os.Stat(core.source)
	if err != nil && !os.IsNotExist(err) {
		return fmt.Errorf("getting file information for %s: %w", core.source, err)
	}
	// in the following code, it's now clear that os.ErrNotExist is non-fatal
	if f != nil && f.IsDir() {
		if uk != nil && uk.UK_BASE != "" {
			core.path = utils.RelativePath(uk.UK_BASE, core.source)
		} else {
			core.path = core.source
		}
	}
```

Reply:
In this scenario, we don't want to set `core.path = core.source` _unless_  `core.source` is a directory. When it is not a directory, the `path` is the location on disk of the extracted microlibrary, e.g. `.unikraft/libs/lwip`. This means that the suggested first error would fail on all build scenarios where the `source:` attribute in the Kraftfile is not a directory.

Reply:
I've added a comment to clarify the reasoning.

Reply:
It wouldn't fail if we handle the error: `if err != nil && !os.IsNotExist(err)`